hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/UpdateStrategy.java

```diff
@@ -0,0 +1,180 @@
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            BIGGER.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    // Only Set support union & intersection
+    UNION {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return combineSet(oldProperty, newProperty, UNION);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            // JsonElements are always List-type, so allows two type now.
+            E.checkArgument((oldProperty instanceof Set ||
+                             oldProperty instanceof List) &&
+                            (newProperty instanceof Set ||
+                             newProperty instanceof List),
+                            formatError(oldProperty, newProperty,
+                                        "Set or List"));
+        }
+    },
+
+    INTERSECTION {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return combineSet(oldProperty, newProperty, INTERSECTION);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    // Batch update Set should use union because of higher efficiency
+    APPEND {
+        @Override
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            ((Collection) oldProperty).addAll((Collection) newProperty);
+            return oldProperty;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    ELIMINATE {
+        @Override
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            ((Collection) oldProperty).removeAll((Collection) newProperty);
+            return oldProperty;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    };
+
+    abstract Object updatePropertyValue(Object oldProperty, Object newProperty);
+
+    abstract void checkPropertyType(Object oldProperty, Object newProperty);
+
+    public Object checkAndUpdateProperty(Object oldProperty,
+                                         Object newProperty) {
+        this.checkPropertyType(oldProperty, newProperty);
+        return this.updatePropertyValue(oldProperty, newProperty);
+    }
+
+    private static String formatError(Object oldProperty, Object newProperty,
+                                      String className) {
+        return String.format("Property must be %s, but got %s, %s", className,
```

Comment:
remove static and make error message more clear, like:

Property type must be %s for strategy %s, but got %s

Reply:
`formatError()`  is referenced by a static method, if we remove static, we have to make it be non-private ?

Reply:
making it protected is ok. I think clear error message is more important.

Reply:
> making it protected is ok. I think clear error message is more important.

done, has unified them with non-static methods,  and since `enum` cannot be inherited ,seems default modifier is enough.